src/java/net/jpountz/lz4/LZ4BlockInputStream.java

```diff
@@ -90,6 +96,9 @@ public LZ4BlockInputStream(InputStream in) {
 
   @Override
   public int available() throws IOException {
+    if (o == originalLen) {
+      refill();
+    }
```

Comment:
I think typically available() just returns the remaining bytes in the buffer, and it should not call a blocking method like refill(), though it is not strictly prohibited. Do you have any reason you want to call refill() in available()?